modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -1135,6 +1147,76 @@ CV__DNN_INLINE_NS_BEGIN
                                    const Scalar& mean = Scalar(), bool swapRB=false, bool crop=false,
                                    int ddepth=CV_32F);
 
+    /**
+     * @brief Enum of image processing mode.
+     * @see ImagePParam
+     */
+    enum ImageProcessMode
+    {
+        DNN_PMODE_CENTER = 0,    // !< Default
+        DNN_PMODE_LETTERBOX = 1, // !< Resize image to the desired size while preserving the aspect ratio of original image.
+    };
+
+    /** @brief Image processing parameters.
+     *
+     * It includes all possible image processing operations and corresponding parameters.
+     *
+     * @see blobFromImageParam
+     * @note
+     * The order and usage of `scalefactor`, `size`, `mean`, `swapRB`, `crop` and `ddepth` are consistent
+     * with the function of @ref blobFromImage.
+    */
+    struct CV_EXPORTS_W_SIMPLE ImagePParam
+    {
+        CV_WRAP ImagePParam();
+        CV_WRAP ImagePParam(const Scalar& scalefactor, const Size& size = Size(), const Scalar& mean = Scalar(),
+                            bool swapRB = false, bool crop = false, int ddepth = CV_32F,
+                            DataLayout datalayout = DNN_LAYOUT_NCHW, ImageProcessMode mode = DNN_PMODE_CENTER);
+
+        CV_PROP_RW Scalar scalefactor; //!< scalefactor multiplier for input image values.
+        CV_PROP_RW Size size;    //!< Spatial size for output image.
+        CV_PROP_RW Scalar mean;  //!< Scalar with mean values which are subtracted from channels.
+        CV_PROP_RW bool swapRB;  //!< Flag which indicates that swap first and last channels
+        CV_PROP_RW bool crop;    //!< Flag which indicates whether image will be cropped after resize or not
```

Comment:
crop flag and ImageProcessMode duplicate each other. Can we remove "crop"?

Reply:
Removed.